test/shell/test_diagnostics_reporter.sh

```diff
@@ -0,0 +1,162 @@
+# shellcheck source=./test_runner.sh
+dir=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+. "${dir}"/test_runner.sh
+. "${dir}"/test_helper.sh
+runner=$(get_test_runner "${1:-local}")
+
+export PATH="$(bazel info workspace)"/tools:$PATH
+
+build_target() {
+  target=$1
+  bazel build --build_event_publish_all_actions //test_expect_failure/diagnostics_reporter:"$target" || true
+}
+
+verify_output() {
+  target=$1
+  expected_outputs=$2
+  execution_root=$(bazel info execution_root)
+  output=$(protoc --decode_raw <"$execution_root"/bazel-out/darwin-fastbuild/bin/test_expect_failure/diagnostics_reporter/"$target".diagnosticsproto)
+  for expected_output in "${expected_outputs[*]}"; do
+    if [[ "$output" != *"$expected_output"* ]]; then
+      echo "Expected: $expected_output, got: $output"
+      exit 1
+    fi
+  done
+
+}
+
+verify_simple_file() {
+  build_target "simple_file"
+  outputs=("1 {
```

Comment:
I'll look into the link you've sent and see what I can do!
Regarding your question, the only way this could break would be if the bazel proto output were to change, or if the scala compiler would change its messages. BSP does not have any control over the structure of a diagnostics, since it uses the [LSP definition](https://microsoft.github.io/language-server-protocol/specification#diagnostic)

Reply:
I was reading about golden master testing and figured that a way of doing so would probably be to commit either the binaries or the decoded version of the diagnostics and compare them. Is this what you meant? And which version would be preferable? I'm assuming the decoded one but would appreciate to hear your input.

Reply:
I've thought about it a bit more- Why are we comparing the full string?
Is it mandatory for us to cover the logic introduced here?
If it was just the easy or intuitive way to do it then I'd like to ask wdyt about changing the tests to be more fuzzy (change to contains of X, Y and Z vs equals)?
The motivation is to make these tests to be more readable and much less brittle

Reply:
I've thought of this with my mentor as well, but I don't know if that would be doable in bash, would I be able to query for specific parameters using `protoc`? Cause if so, that's probably a much better solution

Reply:
1. I don't know protoc well enough but I agree that would be great. Maybe we can run a small scala binary that gets the proto and does assertion (if from scala you have better mechanisms)?
2. What I meant was just to have fuzzier assertions (1 equals => n contain)

Reply:
Using scala, I can definitely do it yes, I'll do some investigation if I'm able to do it in bash and use scala if that turns out not to be possible!

Reply:
Turns out to use the proto libraries with scala, it is needed a new dependency (during compilation it fails to find anything related to the proto library), so I wrote it in java. 
Me and my mentor also discussed that it would probably be best not to compare the error message since that is more prone to changes. Let me know what you think